python/ray/workers/setup_runtime_env.py

```diff
@@ -101,3 +111,69 @@ def setup(input_args):
     command_separator = " && "
     command_str = command_separator.join(commands)
     os.execvp("bash", ["bash", "-c", command_str])
+
+
+def current_ray_pip_specifier() -> Optional[str]:
+    """The pip requirement specifier for the running version of Ray.
+
+    Returns a string which can be passed to `pip install` to install the
+    currently running Ray version.  Returns None if running on a version
+    built from source locally (likely if you are developing Ray).
+
+    Examples:
+        "1.4.0" if running the stable release
+        "https://s3-us-west-2.amazonaws.com/ray-wheels/master/[...].whl" if
+            running the nightly or a specific commit
+    """
+    if os.environ.get("RAY_CI_POST_WHEEL_TESTS"):
+        # Running in Buildkite CI after the wheel has been built.
+        return os.path.join("/ray/.whl", get_wheel_filename())
+    elif ray.__commit__ == "{{RAY_COMMIT_SHA}}":
+        # Running on a version built from source locally.
+        return None
+    elif "dev" in ray.__version__:
+        # Running on a nightly wheel.
+        return (f"https://s3-us-west-2.amazonaws.com/ray-wheels/master/"
+                f"{ray.__commit__}/{get_wheel_filename()}")
+    else:
+        return f"ray=={ray.__version__}"
```

Comment:
Does this need to be `ray[full]` or something?  

What about for the wheel links, are all the sub-libraries included in the wheels by default?

Reply:
I wonder if there's a way to figure out which ones are currently installed to insert them properly

Reply:
For now the best solution might just be `ray[all]`